JS/Lesson34/gulcin/recipe-app/src/App.js

```diff
@@ -1,23 +1,35 @@
 import { useState, useEffect } from 'react';
 import { Search } from './modules/recipes/SearchBar';
+import { RecipeDetails } from './modules/recipes/RecipeDetails';
 import './App.css';
 
 export const App = () => {
 	const [recipes, setRecipes] = useState([]);
+	const [id, setId] = useState('');
 
-	useEffect(() => {
+	function handleClick(e) {
```

Comment:
There's a common patter for it in react: 
```
const handleClick = (id) => setId(id);
```

In fact, you don't even need an external function in this case, just do like this:
```
{recipes.map((recipe => <li key={recipe.idMeal} id={recipe.idMeal} onClick={() => setId(recipe.idMeal)}>{recipe.strMeal}</li>))}
```

I don't think that you need to keep id of the <li> after that, so feel free to remove it too.

Reply:
done